src/goto-cc/compile.cpp

```diff
@@ -634,29 +635,32 @@ bool compilet::write_bin_object_file(
 /// \return true on error, false otherwise
 bool compilet::parse_source(const std::string &file_name)
 {
-  if(parse(file_name))
+  language_filest language_files;
+  language_files.set_message_handler(get_message_handler());
+
+  if(parse(file_name, language_files))
     return true;
 
-  if(typecheck()) // we just want to typecheck this one file here
+  // we just typecheck one file here
+  if(language_files.typecheck(symbol_table))
+  {
+    error() << "CONVERSION ERROR" << eom;
```

Comment:
Doesn't `typecheck` also print the "CONVERSION ERROR" message?

Reply:
No; that's either the language_ui code (getting replaced here) or the two generators of goto_models in goto-programs.